djautotask/sync_rest.py

```diff
@@ -271,3 +291,130 @@ def _assign_field_data(self, instance, json_data):
+        instance.service_level_agreement_has_been_met = \
+            bool(json_data.get('serviceLevelAgreementHasBeenMet'))
+        sla_paused = \
+            json_data.get('serviceLevelAgreementPausedNextEventHours')
+        instance.first_response_date_time = \
+            json_data.get('firstResponseDateTime')
+        instance.first_response_due_date_time = \
+            json_data.get('firstResponseDueDateTime')
+        instance.resolution_plan_date_time = \
+            json_data.get('resolutionPlanDateTime')
+        instance.resolution_plan_due_date_time = \
+            json_data.get('resolutionPlanDueDateTime')
+        instance.resolved_date_time = \
+            json_data.get('resolvedDateTime')
+        instance.resolved_due_date_time = \
+            json_data.get('resolvedDueDateTime')
+        instance.create_date = json_data.get('createDate')
+        instance.due_date_time = json_data.get('dueDateTime')
+        instance.completed_date = json_data.get('completedDate')
+        instance.last_activity_date = json_data.get('lastActivityDate')
+
+        instance = self._parse_date_time(instance, 'first_response_date_time')
+        instance = self._parse_date_time(instance,
+                                         'first_response_due_date_time')
+        instance = self._parse_date_time(instance, 'resolution_plan_date_time')
+        instance = self._parse_date_time(instance,
+                                         'resolution_plan_due_date_time')
+        instance = self._parse_date_time(instance, 'resolved_date_time')
+        instance = self._parse_date_time(instance, 'resolved_due_date_time')
+        instance = self._parse_date_time(instance, 'create_date')
+        instance = self._parse_date_time(instance, 'due_date_time')
+        instance = self._parse_date_time(instance, 'completed_date')
+        instance = self._parse_date_time(instance, 'last_activity_date')
+
+        udfs = json_data.get('userDefinedFields')
+
+        # Refresh udf field to eliminate stale udfs
+        instance.udf = dict()
+
+        if len(udfs):
+            self._assign_udf_data(instance, udfs)
+
+        if sla_paused:
+            instance.service_level_agreement_paused_next_event_hours = \
+                Decimal(str(round(sla_paused, 2)))
+        if instance.estimated_hours:
+            instance.estimated_hours = \
+                Decimal(str(round(instance.estimated_hours, 2)))
+
+        self.set_relations(instance, json_data)
+        return instance
+
+    def get_page(self, next_url=None, *args, **kwargs):
+        kwargs['or_items'] = self.api_conditions
+        return self.client.get_tickets(next_url, *args, **kwargs)
+
+    def get_single(self, ticket_id):
+        return self.client.get_ticket(ticket_id)
+
+    def fetch_sync_by_id(self, instance_id):
+        instance = super().fetch_sync_by_id(instance_id)
+        if instance.completed_date > parse(self.completed_date):
```

Comment:
Is it a replacement for `if instance.status.id != models.Status.COMPLETE_ID:`? If so, I don't see how comparing completed_dates is better than checking for a closed status.

Reply:
I still don't understand this. Can you explain it to me a bit?

Reply:
Ooops! Is that sitll there? Just a min. It should be changed.... Can't you see the change

![image](https://user-images.githubusercontent.com/73486314/115940753-955de200-a457-11eb-8a80-4fa8d26391ca.png)
```
    def fetch_sync_by_id(self, instance_id):
        instance = super().fetch_sync_by_id(instance_id)
        if instance.status.id != models.Status.COMPLETE_ID:
            self.sync_related(instance)
        return instance
```

Reply:
Yes, it was changed:

![image](https://user-images.githubusercontent.com/73486314/115940839-cdfdbb80-a457-11eb-8d7d-67f2645ae6b4.png)

Reply:
Oops! My mistake. I had the wrong commit selected. Sorry about that. Thanks for clarifying